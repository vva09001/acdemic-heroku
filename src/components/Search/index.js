import React, { useCallback, useState } from "react";
import icon from "../../assets/image/Search.svg";

const Search = ({ onKeyPressSearch, onClickSearch, placeholder }) => {
  const [searchQuery, setSearchQuery] = useState("");

  const handleKeyPress = useCallback(e => {
    if (e.which === 13) {
      onKeyPressSearch && onKeyPressSearch(searchQuery);
      setSearchQuery("");
    }
  });

  const handleClick = useCallback(e => {
    onClickSearch && onClickSearch(searchQuery);
    setSearchQuery("");
  });
  return (
    <React.Fragment>
      <div className="search">
        <img onClick={e => handleClick(e)} src={icon} alt="" />
        <input
          value={searchQuery}
          onChange={e => setSearchQuery(e.target.value)}
          placeholder={placeholder}
          onKeyPress={e => handleKeyPress(e)}
        />
      </div>
    </React.Fragment>
  );
};

export default Search;
