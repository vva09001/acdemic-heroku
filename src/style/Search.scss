$primaryColor: #203e7d;
$secondColor: #0060ae;

.search {
  padding: 5px;
  display: flex;
  box-shadow: 0 8px 6px -6px #777;
  align-items: center;
  border-radius: 2px;
  background: #f3f3f3;
  min-height: 35px;
  img {
    height: 20px;
  }
  input {
    background: transparent;
    margin-left: 5px;
    width: 100%;
    font-size: 12px;
    box-shadow: none;
    padding: 0;
    &::placeholder {
      font-style: italic;
      color: #777;
    }
  }
}

.searchQuery {
  margin-top: 10px;
  font-size: 12px;
  color: $primaryColor;
  font-weight: bold;
}