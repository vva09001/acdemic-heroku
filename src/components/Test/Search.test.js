import React from "react";
import { create } from "react-test-renderer";
import Search from "../Search";

describe("SearchComponent", () => {
  it("Test Search component it work", () => {
    const component = create(<Search />).toJSON();
    expect(component).toMatchSnapshot();
  });
  exports[`Rerender Search component`] = (
    <Search
      onKeyPressSearch={Function}
      onClickSearch={[Function]}
      placeholder={[String]}
    />
  );
});
