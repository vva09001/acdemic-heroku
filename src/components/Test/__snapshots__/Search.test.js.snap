// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent Test Search component it work 1`] = `
<div
  className="search"
>
  <img
    alt=""
    onClick={[Function]}
    src="Search.svg"
  />
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    value=""
  />
</div>
`;
